### PR TITLE
Bump to the dev version when a new patch branch is created

### PR DESCRIPTION
**Pull request by waiting-for-dev**
## Summary

When there's an existing patch branch (e.g., `v3.4`, `v4.0`...) and a new release is made, we're already [bumping to the `.dev` version for the next patch release](https://github.com/solidusio/solidus/blob/3d9acfcc83cc67d91db6e1573247a14b28329d75/.github/workflows/prepare_post_release.yml#L29).

However, when a new [patch branch is created](https://github.com/solidusio/solidus/blob/3d9acfcc83cc67d91db6e1573247a14b28329d75/.github/workflows/prepare_post_release.yml#L62) we were missing this. For instance, after releasing `v4.0.0` the newly created `v4.1` branch would still point at `4.0.0` instead of `4.0.1.dev` in `Spree::VERSION`.

That wasn't a problem for minor releases, but it caused issues when a patch branch looked like tracking a next major release (like the given `4.0.0` example) (see #5108).

We're fixing this by creating another Pull Request for the new created patch branch, bumping to the next patch `.dev` version. To allow this, we're modifying the pull request action to allow specifying a base branch different than the one running the workflow, and allowing a workflow to create more than one pull request by adding a timestamp to the branches it generates.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

**Review by elia — APPROVED** [review]
Approving, although I don't have a full grasp of what is going on here, trusting it was tried out and also being dev-tooling can be easily fixed.

File: dev_tools/lib/solidus/pipeline_context.rb [hunk 1]
```diff
@@ -69,6 +69,18 @@ def candidate_minor_version
       @candidate_minor_version ||= candidate_version.split(VERSION_SEPARATOR)[0..1].join(VERSION_SEPARATOR)
     end
 
+    def candidate_next_patch_dev_version
+      @candidate_next_patch_dev_version ||=
+        begin
```
[the remark this answers is not in the document]
Comment (waiting-for-dev) [reply dev_tools/lib/solidus/pipeline_context.rb]: True. Fixed. Thanks!

**Comment by waiting-for-dev** [comment]
> Approving, although I don't have a full grasp of what is going on here, trusting it was tried out and also being dev-tooling can be easily fixed.

Yep, it has been tested in a personal fork. We have some [documentation](https://github.com/solidusio/solidus/wiki/Release-pipeline-automation) in the wiki about all the release automation, but happy to explain anything or walk you through it!

**Review by kennyadsl — APPROVED** [review]
Thanks Marc!

**Pull request merged**